lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/persistence/DBOTeamTest.java

```diff
@@ -83,4 +89,54 @@ public void testRoundTrip() throws DatastoreException, NotFoundException, Unsupp
 		assertEquals(clone, clone2);
 	}
 
+	@TemporaryCode(author = "peter.harvey@sagebase.org", comment = "One time migration of state.  Can be removed after all teams have a state value.")
+	@Test
+	public void testCreateDatabaseObjectFromBackupWithPublicTeam() {
+		Team dto = new Team();
+		dto.setId("123");
+		dto.setEtag(UUID.randomUUID().toString());
+		dto.setCanPublicJoin(true);
+		dto.setCanRequestMembership(false);
+		DBOTeam dbo = new DBOTeam();
+		TeamUtils.copyDtoToDbo(dto, dbo);
+
+		// Call under test
+		dbo = dbo.getTranslator().createDatabaseObjectFromBackup(dbo);
+
+		assertEquals(TeamState.PUBLIC.name(), dbo.getState());
+	}
+
+	@TemporaryCode(author = "peter.harvey@sagebase.org", comment = "One time migration of state.  Can be removed after all teams have a state value.")
+	@Test
+	public void testCreateDatabaseObjectFromBackupWithOpenTeam() {
+		Team dto = new Team();
+		dto.setId("123");
+		dto.setEtag(UUID.randomUUID().toString());
+		dto.setCanPublicJoin(false);
+		dto.setCanRequestMembership(true);
+		DBOTeam dbo = new DBOTeam();
+		TeamUtils.copyDtoToDbo(dto, dbo);
+
+		// Call under test
+		dbo = dbo.getTranslator().createDatabaseObjectFromBackup(dbo);
+
+		assertEquals(TeamState.OPEN.name(), dbo.getState());
+	}
+
+	@TemporaryCode(author = "peter.harvey@sagebase.org", comment = "One time migration of state.  Can be removed after all teams have a state value.")
+	@Test
+	public void testCreateDatabaseObjectFromBackupWithClosedTeam() {
+		Team dto = new Team();
+		dto.setId("123");
+		dto.setEtag(UUID.randomUUID().toString());
+		dto.setCanPublicJoin(false);
+		dto.setCanRequestMembership(false);
+		DBOTeam dbo = new DBOTeam();
+		TeamUtils.copyDtoToDbo(dto, dbo);
+
+		// Call under test
+		dbo = dbo.getTranslator().createDatabaseObjectFromBackup(dbo);
+
+		assertEquals(TeamState.CLOSED.name(), dbo.getState());
+	}

```

Comment:
We do need a test where the state is already set, to make sure we are not re-setting each time